In [243]:
import pandas as pd
from sklearn.preprocessing import StandardScaler
import numpy as np

df = pd.read_csv("data/imputed_dataset.csv")

In [244]:
df = df[(df['fyear'] >= 2000) & (df['fyear'] <= 2023)]
df = df.dropna(subset=['prev_inv'])

In [245]:
X = df[['liquidity', 'leverage', 'roa', 'icapt', 'aqc', 'prev_inv', 'cpu_index', 'asset_growth']]
y = df['y']

In [246]:
X = X.applymap(lambda x: np.log(x) if x > 0 else x)
y = np.log(df['y'])

In [247]:
scaler = StandardScaler()
scaler.fit(X)
X_scaled = scaler.transform(X)

In [248]:
X.head()

liquidity  leverage       roa     icapt       aqc  prev_inv  cpu_index  \
276  -0.405825 -0.693197 -3.163223  8.375122  0.000000 -2.037374   4.134346   
277   0.540221 -0.690331 -1.702159  9.174431  0.000000 -2.904557   4.134346   
278   0.775244 -0.952044 -1.769399  8.375547  0.000000 -2.118500   4.134346   
279   0.272252 -0.718890 -4.198557  8.622022  5.138149 -2.434787   4.134346   
280   0.390569 -0.296824 -2.719636  9.521348  7.833204 -3.319816   4.134346   

     asset_growth  
276      0.078636  
277      0.054608  
278      0.275756  
279      0.004241  
280      0.067703

In [249]:
X.isnull().sum()

liquidity       0
leverage        0
roa             0
icapt           0
aqc             0
prev_inv        0
cpu_index       0
asset_growth    0
dtype: int64

# Model 1 without rnd

In [250]:
import statsmodels.api as sm

X = sm.add_constant(X)

model = sm.OLS(y, X).fit()

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.728
Model:                            OLS   Adj. R-squared:                  0.727
Method:                 Least Squares   F-statistic:                     2116.
Date:                Fri, 08 Mar 2024   Prob (F-statistic):               0.00
Time:                        13:54:34   Log-Likelihood:                -3569.4
No. Observations:                6347   AIC:                             7157.
Df Residuals:                    6338   BIC:                             7218.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            0.4519      0.068      6.633   

In [251]:
# Check instrument relevance (F-statistic)
f_statistic = model.fvalue
if f_statistic < 10:
    print("Warning: Instruments may not be relevant.")
else: 
    print("Significant Instruments")

Significant Instruments


**t-values**:
   - t-values represent the ratio of the estimated coefficient to its standard error.
   - They indicate the significance of the estimated coefficient.
   - Larger absolute t-values (in either direction) suggest greater evidence against the null hypothesis that the coefficient is zero.
   
**P-values (P>|t|)**:
   - P-values represent the probability of observing the estimated coefficient (or more extreme) if the null hypothesis (that the coefficient is zero) is true.
   - Small p-values (typically less than a chosen significance level, such as 0.05) suggest that the coefficient is statistically significant.
   - In this output, all p-values are very close to zero (0.000), indicating that all coefficients are statistically significant at conventional significance levels (e.g., 0.05).
   
**Confidence Intervals ([0.025 0.975])**:
   - Confidence intervals provide a range of plausible values for the true population parameter (the coefficient) based on the sample data.
   - They are typically constructed around the estimated coefficient and provide a sense of the precision of the estimate.
   - The confidence interval is calculated as the estimated coefficient plus or minus a margin of error.
   - In this output, the confidence intervals indicate the range within which we are reasonably confident the true coefficients lie.

**R-squared (or Adjusted R-squared) Test:**
R-squared measures the proportion of variance in the dependent variable that is explained by the independent variables in the model.
Adjusted R-squared adjusts for the number of independent variables in the model, providing a more conservative measure of model fit.
Both R-squared and Adjusted R-squared range from 0 to 1, with higher values indicating better fit.

In [252]:
# Calculate R-squared
r_squared = model.rsquared
print("R-squared:", r_squared)

# Calculate Adjusted R-squared
adjusted_r_squared = model.rsquared_adj
print("Adjusted R-squared:", adjusted_r_squared)


R-squared: 0.7275620320910468
Adjusted R-squared: 0.7272181533054248


The F-test assesses the overall significance of the model by comparing the explained variance to the unexplained variance.
A significant F-test indicates that at least one independent variable has a nonzero coefficient, suggesting that the model as a whole is significant.

In [253]:
# Get F-statistic and associated p-value
f_statistic = model.fvalue
p_value = model.f_pvalue

print("F-statistic:", f_statistic)
print("p-value:", p_value)


F-statistic: 2115.7514290253553
p-value: 0.0


# Model 2 with R&D

In [254]:
df['xrd'].fillna(0, inplace=True)
X2 = df[['liquidity', 'leverage', 'roa', 'icapt', 'aqc', 'prev_inv', 'cpu_index', 'asset_growth', 'xrd']]

In [255]:
X2 = sm.add_constant(X2)

model2 = sm.OLS(y, X2).fit()

print(model2.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.556
Model:                            OLS   Adj. R-squared:                  0.555
Method:                 Least Squares   F-statistic:                     881.2
Date:                Fri, 08 Mar 2024   Prob (F-statistic):               0.00
Time:                        13:54:34   Log-Likelihood:                -5120.4
No. Observations:                6347   AIC:                         1.026e+04
Df Residuals:                    6337   BIC:                         1.033e+04
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           -3.6190      0.024   -147.926   

# Model 3 with change in investment

In [256]:
columns_to_difference = ['liquidity', 'leverage', 'roa', 'icapt', 'aqc', 'asset_growth', 'y']
df_differences = df[columns_to_difference].diff()

df_differences.columns = [col + '_delta' for col in columns_to_difference]

df = pd.concat([df, df_differences], axis=1)
df = df.dropna()
print(df)

      Unnamed: 0   GVKEY    datadate  fyear indfmt consol popsrc datafmt  \
277          277    1078  31/12/2000   2000   INDL      C      D     STD   
278          278    1161  31/12/2000   2000   INDL      C      D     STD   
279          279    1209  30/09/2000   2000   INDL      C      D     STD   
280          280    1300  31/12/2000   2000   INDL      C      D     STD   
281          281    1327  31/03/2001   2000   INDL      C      D     STD   
...          ...     ...         ...    ...    ...    ...    ...     ...   
6617        6617  114524  31/12/2022   2022   INDL      C      D     STD   
6618        6618  117768  31/01/2023   2022   INDL      C      D     STD   
6620        6620  121077  30/09/2022   2022   INDL      C      D     STD   
6621        6621  121718  31/12/2022   2022   INDL      C      D     STD   
6622        6622  125595  31/12/2022   2022   INDL      C      D     STD   

                              conm curcd  ...  asset_growth  year   cpu_index  \
277   

In [257]:
X3 = df[['liquidity_delta', 'leverage_delta', 'roa_delta', 'icapt_delta', 'aqc_delta', 'cpu_index', 'asset_growth_delta']]
y = df['y']

In [258]:
X3.isnull().sum()

liquidity_delta       0
leverage_delta        0
roa_delta             0
icapt_delta           0
aqc_delta             0
cpu_index             0
asset_growth_delta    0
dtype: int64

In [259]:
X3 = sm.add_constant(X3)

model3 = sm.OLS(y, X3).fit()

print(model3.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.063
Model:                            OLS   Adj. R-squared:                  0.061
Method:                 Least Squares   F-statistic:                     39.27
Date:                Fri, 08 Mar 2024   Prob (F-statistic):           9.60e-54
Time:                        13:54:34   Log-Likelihood:                 8406.7
No. Observations:                4110   AIC:                        -1.680e+04
Df Residuals:                    4102   BIC:                        -1.675e+04
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
const                  0.0543      0